topo/topo_test.go

```diff
@@ -1944,3 +1944,69 @@ func TestStateMap(t *testing.T) {
 		})
 	}
 }
+
+func TestUpdateServicePortName(t *testing.T) {
+	tests := []struct {
+		desc string
+		key  uint32
+		svc  *tpb.Service
+		want *tpb.Service
+	}{
+		{
+			desc: "services valid names with duplicates",
+			svc: &tpb.Service{
+				Name:   "gnmi",
+				Names:  []string{"gnmi", "gribi", "gribi"},
+				Inside: 9339,
+			},
+			want: &tpb.Service{
+				Name:   "gnmi",
+				Names:  []string{"gnmi", "gribi", "gribi"},
+				Inside: 9339,
+			},
+		},
+		{
+			desc: "services valid names with empty name",
+			svc: &tpb.Service{
+				Names:  []string{"gnmi", "gribi"},
+				Inside: 9339,
+			},
+			want: &tpb.Service{
+				Name:   "gnmi",
+				Names:  []string{"gnmi", "gribi"},
+				Inside: 9339,
+			},
+		},
+		{
+			desc: "services valid names with empty string",
+			svc: &tpb.Service{
+				Name:   "gnmi",
+				Names:  []string{"", "gribi"},
+				Inside: 9339,
+			},
+			want: &tpb.Service{
+				Name:   "gnmi",
+				Names:  []string{"", "gribi"},
```

Comment:
Do we want to keep empty strings?  Is there any reason we should not strip them out?

Reply:
Ack, stripped empty names.